cuML/src/solver/solver.cu

```diff
@@ -340,5 +338,158 @@ void sgdPredictBinaryClass(const double *input, int n_rows, int n_cols,
 
 }
 
+void cdFit(float *input,
+		   int n_rows,
+		   int n_cols,
+		   float *labels,
+		   float *coef,
+		   float *intercept,
+		   bool fit_intercept,
+		   bool normalize,
+		   int epochs,
+		   int loss,
+		   float alpha,
+		   float l1_ratio,
+		   bool shuffle,
+		   float tol) {
+
+	ASSERT(loss == 0,
```

Comment:
Why not use a enum or something else that makes specifying the loss function clearer?

Reply:
I agree. It's near impossible for a user to know what value to set here and an enum could make it immediately obvious (or even better, instead of an enum, make the loss functions pluggable lambas or classes).

Reply:
I use enum after that point but don't know how to make it work with cython. Can you guys help me on that?

Reply:
Used enum after that point but don't know the cython side. Can you guys help for this?

Reply:
Sure. Enums are trivial in cython. I recently used it for UMAP.